dulwich/stash.py

```diff
@@ -62,7 +62,14 @@ def from_repo(cls, repo):
 
     def drop(self, index):
         """Drop entry with specified index."""
-        raise NotImplementedError(self.drop)
+        with open(self._reflog_path, "rb+") as f:
```

Comment:
Wasn't sure if calling `open` directly here is a problem (vs using `GitFile`)? `drop_reflog_entry` needs a read+write file handle, and I noticed that when writing reflog entries for commits in `Repo` open is also used directly (but in append mode)

Reply:
it's good to be consistent. GitFile provides locking, which seems like it would be beneficial. We should double check what Git does here.